clients/ts/signalr/src/LongPollingTransport.ts

```diff
@@ -0,0 +1,163 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+import { AbortController } from "./AbortController";
+import { DataReceived, TransportClosed } from "./Common";
+import { HttpError, TimeoutError } from "./Errors";
+import { HttpClient, HttpRequest } from "./HttpClient";
+import { IConnection } from "./IConnection";
+import { ILogger, LogLevel } from "./ILogger";
+import { ITransport, TransferFormat } from "./ITransport";
+import { Arg, getDataDetail, sendMessage } from "./Utils";
+
+const SHUTDOWN_TIMEOUT = 5 * 1000;
+
+export class LongPollingTransport implements ITransport {
+    private readonly httpClient: HttpClient;
+    private readonly accessTokenFactory: () => string | Promise<string>;
+    private readonly logger: ILogger;
+    private readonly logMessageContent: boolean;
+
+    private url: string;
+    private pollXhr: XMLHttpRequest;
+    private pollAbort: AbortController;
+    private shutdownTimeout: number;
+    private running: boolean;
+
+    constructor(httpClient: HttpClient, accessTokenFactory: () => string | Promise<string>, logger: ILogger, logMessageContent: boolean) {
+        this.httpClient = httpClient;
+        this.accessTokenFactory = accessTokenFactory || (() => null);
```

Comment:
nit: Defaulting to a null string would be a bit more efficient.

Reply:
You mean default to `null` rather than a `null`-returning function? We'd have to change the signature and call site.

Reply:
Ohhhh, I thought it was a string or a func, but it is always a func. nvm